### ISSUE TITLE
App missing from play store.

### ISSUE THREAD

**Issue by ghost** [KonradIT/CamControl#1]
Greetings. I bought recently a hero 5 black ,and tried the original app(the whole registration is annoying).
I couldn't find your app in the store. Is there a way to get/build the app? Have you ever thought about upload it to the fdroid repo?

**Comment by laslanian** [comment]
You could build the app and download the apk file on your phone.

**Comment by exadeci** [comment]
@laslanian  That requires having a full Android development environment set up.

It would be cool if there could be an apk in /releases or to have it on F-Droid or a 3rd party F-Droid repository like https://android.izzysoft.de/repo/info if F-droid rules are too restrictive.

**Comment by KonradIT** [comment]
The app that was published on the play store is https://github.com/konradit/goprohero . The app started to be too complicated to maintain and the UI was too cumbersome, it's my first successful app. I took it down from the play store but now I'm working in this repository to rewrite a spiritual successor to CamControl for GoPro. The new app is independent from Google closed source bits and will work with more cameras. it will be published on F-Droid. Progress of the rewrite is slow due to school, work and life.

**Comment by laslanian** [comment]
@exadeci if you pay attention to his question, he asks whether there is a way to "get/build" the app.

I think my answer addresses this question.

**Comment by ghost** [comment]
@KonradIT thanks for the answer ,and if you even get in beta phase i would love to try i out. And the for me the frontend style outrated, i have seen the screenshots and it was already ok,nothing fancy

**Comment by IzzySoft** [comment]
My repo got pinged, so I got curious :wink:

@KonradIT speaking of "more cameras": I've got a "Sports HD DV" action camera (full HD. Might I have a chance with your app then? In that case I'd keep an eye on it for sure :stuck_out_tongue_winking_eye:

**Comment by KonradIT** [comment]
@IzzySoft Hi, who manufactures the camera and what model is it? The successor (https://github.com/konradit/camcontrol) is similar Gadgetbridge: an open source app that connects to cameras. Find out the API calls, livesteam link and media endpoint and I can add support for it.

**Comment by IzzySoft** [comment]
Thanks @KonradIT – I've [bought it at Amazon](https://www.amazon.de/dp/B06WGNXVKD) last year, not sure about the manufacturer (unfortunately Amazon removed a lot of reviews (5 out of 20 remained) and answered questions, so half the details got lost. According to the bill simply describes it as

> Action Kamera 1,5 Zoll GBB FHD Sport Camera 12MP Wasserdichte 1080P Kamera mit 170°Weitwinkel Sport Action Kamera 900mAh Akku und Zubehör Kitt(Schwarz)

Maybe I should simply load "the successor" and give it a try. With some luck, it's just some "standard China construction" :wink: It came pretty cheap, around 30 Euro. Unfortunately I didn't even find a way to get rid of that "fish eye" 170° …

**Comment by IzzySoft** [comment]
Hm, hard to try without an APK. As @exadeci suggested, coud you provide one in `releases/`? I could then also distribute it via my repo – again as @exadeci suggested :)

**Comment by KonradIT** [comment]
@izzysoft Sure, I'll release them and contact you when I have a first *working* build, which is not that far away. Also if you want to add your camera feel free, I'm looking for cheap Ambarella A7 ones or Novatek based.

**Comment by IzzySoft** [comment]
Looking forward to that, thanks!

As for my cam: no idea what's inside. The so-called "manual" provided needs a microscope to read (and pencils to hold), plus is very basic. But hey, what to expect for 30 Euros? If it works with your app, I'll see to provide as much details as possible so you can list it.

**Comment by KonradIT** [comment]
@IzzySoft your action camera has a different WiFi API than GoPros, so for the app to work you would need to sniff the commands from your camera's original app and send them to me so I can add your camera to the app.

**Comment by IzzySoft** [comment]
@KonradIT to be honest, I never found that "original app" (I tried, but didn't succeed). And even if I had it, I wouldn't know how to "sniff the commands".

You wrote that, apart from the GoPros, your app already supports some other "China cams" (AFAIR that is). Considering the price, I'd assume mine is some such "China clone". So I'd say I just wait until you have an APK ready and try that. If it doesn't work, we still can see what to do about it. It's nothing urgent for me (at least not at the moment) as I rarely use the cam (bought it mainly for holiday and diving, as it ships with a case water proof to 3atm / 30m).

**Comment by rexxxx1234** [comment]
@KonradIT Thank you for your great work! Just a quick question, I downloaded, built and ran the app in Android Studio and it seems that the GoPro APIs have not been integrated into the app yet(all the API codes are there). It this something you are working on?

**Comment by KonradIT** [comment]
@rexxxx1234 yes, I work basically full time now so I don't have much time for development. I'm still designing a suitable universal structure for all camera APIs so it's easy to integrate new cameras.

**Comment by rexxxx1234** [comment]
@KonradIT I ran the app on a Tablet with PreviewActivity and tried to get the live streaming to work. I connected the Tablet with GoPro Hero4 through wifi. But it is having trouble with createPlayer and stream fails. Is there any step I missed?

**Comment by KonradIT** [comment]
@rexxxx1234 app is not working right now.

**Comment by IzzySoft** [comment]
> when I have a first working build, which is not that far away.

Got any closer, @KonradIT? I'm pretty sure it would work with my camera – after I found the following file on its card:

```json
{
"info version":"2.0",
"firmware version":"HD4.01.05.00.00",
"wifi mac":"<sanitized>",
"camera type":"HERO4 Silver",
"camera serial number":"<sanitized>",
}
```

You see it's pretending to be a Hero4 (though it doesn't have the front-facing display, and probably lacks some other things – but I wouldn't dare trying a firmware upgrade or I might brick it). So if you have an APK I could try, I'd give it a look :smiley:

**Comment by KonradIT** [comment]
@IzzySoft neat! I've been deadlocked for a while on this project. My main focus has been updating my python API wrapper and keeping docs up to date. I have new responsibilities at work and over time been allocating less time for open source projects. But I did some progress as the new HERO8 camera should be supported by the app as well as GoPro MAX (I have it, and will add support for it). So far cameras supported by me: HERO4 Black, HERO5 Black, HERO7 Black, MAX, Session, HERO3/3+. Hopefully next year I get some more time off and can focus on this app. Unfortunately my Android skills are very dated, so will likely take a course to refresh on the new stuff.

**Comment by IzzySoft** [comment]
Thanks for the heads-up, @KonradIT! Btw: after my enthusiastism on above find came a disillusion: no WiFi on that cam it seems – despite of the settings showing a MAC (maybe they just disabled it, and I just didn't find the "easter egg" to enable it – like, "use morse to write 'wifion': down is long, up is short" :see_no_evil:).

I was hoping to get a bit easier access to the cam's settings (it should somehow be possible to reduce the 170° fish-eye to something like 130° wide – but the internal settings don't offer that). So for now, it seems I'll have to enjoy USB mass storage only.

Well, at that price, I cannot even complain. Picture quality is fine, it makes full HD vids, comes with a case for diving – and all that for less than 50 bucks. :man_shrugging:

**Comment by KonradIT** [comment]
@IzzySoft Get one of the older GoPros, HERO5 or HERO6. HERO4 is still good and can be found for less than 100.00 EUR

**Comment by IzzySoft** [comment]
Thanks Konrad. But as rarely as I use the cam (mostly just for diving once a year if my better half permits me), those "missing features" are rather "icing on the cake" (ugh, and it seems WiFi can be made working: power-on, then "up", password "12345678"). As long as the cam works, I'll stick with it as-is. If I can get more out of it, great – but no need to buy another one for that :wink:

Btw: I'm still open to host your app in my repo, as suggested by Exadeci above – so feel free to ping me if there's a release with APK attached.

**Comment by WPFilmmaker** [comment]
+1 for publishing it on f-droid!
